File: LC1773.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        int k, c = 0;

        if(ruleKey == "type")
            k = 0;
        else if(ruleKey == "color")
            k = 1;
        else
            k = 2;

        for(auto i : items)
            if(i[k] == ruleValue)
                c++;
        
        return c;
    }
};

File: LC1773.py
class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        d = {"type" : 0, "color" : 1, "name" : 2}
        
        c = 0

        for i in items:
            c += (1 if i[d[ruleKey]] == ruleValue else 0)
        
        return c
